[PJRT]  Expose should_stage_host_to_device_transfers as client create option

hugomano opened the pull request:
Expose GPU option `should_stage_host_to_device_transfers ` as configurable in PJRT client create function.
This allow the end user to override the default value which is `true`.

ref: [openxla@xla/blob/main/xla/pjrt/plugin/xla_gpu/xla_gpu_client_options.h](https://github.com/openxla/xla/blob/main/xla/pjrt/plugin/xla_gpu/xla_gpu_client_options.h)

Since the option is living in a hashmap of type PJRT_NamedValue, it seems that I don't break the C PJRT API interface versioning.

changm commented:
Sorry for the delay here, I was on vacation last week :). Any idea if there's some test that could detect if this options was plumbed correctly?

hugomano commented:
Hey @changm thanks for your review. 
There was no test regarding `should_stage_host_to_device_transfers ` so I added them in `StreamExecutorGpuClientTest` and `PjrtCApiGpuExtensionTest`.

changm commented on xla/pjrt/c/pjrt_c_api_gpu_test.cc:
```diff
@@ -689,6 +690,67 @@ TEST(PjrtCApiPlatformNameTest, UnavailablePlatformName) {
   api->PJRT_Error_Destroy(&error_destroy_args);
 }
 
+TEST(PjrtCApiGpuExtensionTest, ShouldStageHostToDeviceTransferOption) {
```
nit: Break this out into two separate tests, one enabled and one disabled.

changm commented on xla/pjrt/c/BUILD:
```diff
@@ -442,6 +442,7 @@ xla_test(
         "//xla/pjrt:pjrt_common",
         "//xla/pjrt:pjrt_compiler",
         "//xla/pjrt:pjrt_future",
+        "//xla/pjrt/gpu:se_gpu_pjrt_client",
```
nit: Can you rely on the public API instead?

https://github.com/openxla/xla/tree/main/xla/pjrt/plugin/xla_gpu

changm commented on xla/pjrt/gpu/se_gpu_pjrt_client_test.cc:
```diff
@@ -1118,6 +1118,61 @@ TEST(StreamExecutorGpuClientTest, MockNcclClientTest) {
+  TF_ASSERT_OK_AND_ASSIGN(auto client_staging, 
+                          GetStreamExecutorGpuClient(options_staging));
+
+  GpuClientOptions options_no_staging;
```
ditto: Can you please break this up into two different tests instead?

changm reviewed (approved):
LGTM assuming you make the changes in the comments. Thanks!

hugomano commented:
Thanks @changm I addressed your comments. Can you approve the CI workflow (I only tested locally through `tensorflow/ml-build:latest`) to check if I missed something?

changm commented:
I think I approved the internal one as well. Looping in @golechwierowicz as build rotation. Can you help unjam this? Or do I need to get write approval somewhere?

jparkerh reviewed (approved):
Approving to trigger a re-import to copybara, seems the original change is gone.

/edit, I missed the change, this is still in review internally.
